### ISSUE TITLE
Local config

### ISSUE THREAD

**Issue by JulioC** [carloscuesta/gitmoji-cli#59]
Hello @carloscuesta!

Currently I couldn't find a way to set the config specific for a project or based on command line flags.

It would be useful to have that so one could create standard settings for a given project, across all users.

**Comment by arnochauveau** [comment]
@JulioC i've found a temporary workaround:

You can specify your own config via the `NODE_CONFIG` environment variable.
You might need to search a bit in the source code to find the right options. `config.js` and `constants.js` should contain all the info you need. 

for example:
`NODE_CONFIG='{"autoAdd": "true"}' gitmoji -c`

I would recommend making an alias for this command :smile:

**Comment by Addono** [comment]
Gave this one a shot, progress can be found [here](https://github.com/Addono/gitmoji-cli/tree/feature/local_config). Not really happy about it, as detecting the current git repository is done async, hence retrieving `config` from `configurationVault` becomes async changing the interface it exposes. 

In my attempt I assumed that it would be undesirable to use another library for config handling, as this would either require supporting migrations from the old to the new one, or running them side-by-side causing an increase in complexity.

Right now it lacks:
- [ ] Indication whether local configuration files are used*
- [ ] Support for enabling/disabling local config from the CLI**

Alternatively of storing the config in the `.git` folder, I was thinking about adding all local config in the global configuration and storing some identifier/token in the pre-commit hook. But that would not work for a direct `gitmoji -c` invocation.

Anyway, I am not sure if this is the way to go, or if it is better to wait for top-level awaits to land and use that. Now a relatively small feature has quite some impact on the overall code base 😒.

For now, not having this feature is not really a deal-breaker for me, there's only one project where the scope prompt would be nice, everything else just uses the global config.

*Had this in a [previous version](https://github.com/Addono/gitmoji-cli/commit/932ddb37afa8fbf5493b1e505a39f0fec25e26a4), where I exposed an additional function `configurationVault:isLocal` which printed whether you are using local config when invoking `gitmoji --config`.

**For now, this needs to be done manually by initializing the local config yourself, e.g. run this in the root of your Git project:
```bash
echo '{"autoAdd": true, "emojiFormat": "emoji", "signedCommit": false, "scopePrompt": false}' > .git/.gitmoji/config.json
```

**Comment by carloscuesta** [comment]
This is a great idea! I'm not sure if it's really valuable for the cli since we do not have a lot of configuration options however I think it's not that difficult to implement.

Priorities for configuration should be obtained from:

- `gitmoji -g` config params
- gitmojirc local file (not using .dotfile format)
- `gitmoji` field on a package.json

**Comment by ooga** [comment]
Minimal support https://github.com/carloscuesta/gitmoji-cli/pull/679

**Comment by carloscuesta** [comment]
👋🏼 Hello everyone

I implemented this in #901, from `gitmoji-cli@6.0.0` you'll be able to specify the configuration parameters through the `package.json`, a `.gitmojirc.json` file or the cli options:

Read more about this in the docs: https://github.com/carloscuesta/gitmoji-cli#config
